backend/parser.py

```diff
@@ -1,45 +1,64 @@
+"""
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
 import re
 
 class Sample:
     """Sample holds the data recorded for some sensor.
 
     Attributes:
         sensor_name: A string holding the name of the sensor.
-        sensor_id: The numerical i.d. of the sensor.
+        sensor_id: The unique ID of a sensor.
         timestamps: A list of each recorded timestamp in the sample.
         timestamp_diffs: A list where index i contains timestamps[i] - timestamps[i-1]. If i==0, timestamps_diff[i] = 0.
-        data: A dictionary that is initialized to hold 3D data but can be extended to any D data.
-            e.g. data[0] holds a list of all the data's x points
+        data: A dictionary that holds a list of each dimmensions recorded datapoints. Initialized to hold 1D data,
+            But can be extended to hold an arbitary number of dimmensions.
+                e.g. data[0] holds a list of all the data's first dimmension datapoints, usually the x-axis.
         latencies: A List of the recorded latency of each datapoint
         next_index: Index where the next datapoint will be added for every list in the sample
     """
 
-    def __init__(self, sensor_name: str, sensor_id):
+    def __init__(self, sensor_name: str, sensor_id: str):
         """Initializes Sample with the sensor_name and a numeric sensor_id"""
         self.sensor_name = sensor_name
         self.sensor_id = sensor_id
 
         #tracks the next index where data will be added
```

Comment:
Comments should have a space between the '#' and the text. Comments should also follow basic grammer rules (capitalization, punctuation, etc):

# Tracks the next index where data will be added.

https://google.github.io/styleguide/pyguide.html#385-block-and-inline-comments

This comment also applies to other lines in this pull request.